bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -139,6 +141,20 @@ def initialize(lockfile)
         end
         @pos.advance!(line)
       end
+
+      if !Bundler.frozen_bundle? && @platforms.include?(Gem::Platform::X64_MINGW_LEGACY)
+        if @platforms.include?(Gem::Platform::X64_MINGW)
+          SharedHelpers.major_deprecation(2,
+            "Found x64-mingw32 in lockfile, which is deprecated. Ignoring it. Support for x64-mingw32 will be removed in Bundler 3.0.",
+            removed_message: "Found x64-mingw32 in lockfile, which is no longer supported as of Bundler 3.0. Ignoring it.")
+        else
+          @platforms[@platforms.index(Gem::Platform::X64_MINGW_LEGACY)] = Gem::Platform::X64_MINGW
+          SharedHelpers.major_deprecation(2,
+            "Found x64-mingw32 in lockfile, which is deprecated. Using the current x64-mingw-ucrt platform instead. Support for x64-mingw32 will be removed in Bundler 3.0.",
+            removed_message: "Found x64-mingw32 in lockfile, which is no longer supported as of Bundler 3.0. Using the current x64-mingw-ucrt platform instead.")
```

Comment:
Like the other `removed_message`, don't mention "Using the current x64-mingw-ucrt platform instead.", because Bundler invokation will fail with an error.